src/utils/export/index.ts

```diff
@@ -77,3 +93,70 @@ export const getGoogleLikeMetadataFile = (
         2
     );
 };
+
+export const sanitizeName = (name: string) =>
+    name.replaceAll('/', '_').replaceAll(' ', '_');
+
+export const getUniqueCollectionFolderPath = (
+    dir: string,
+    collectionName: string,
+    usedCollectionPaths: Set<string>
+): string => {
+    let collectionFolderPath = `${dir}/${sanitizeName(collectionName)}`;
+    let count = 1;
+    while (usedCollectionPaths.has(collectionFolderPath)) {
+        collectionFolderPath = `${dir}/${sanitizeName(
+            collectionName
+        )}(${count})`;
+        count++;
+    }
+    usedCollectionPaths.add(collectionFolderPath);
+    return collectionFolderPath;
+};
+
+export const getMetadataFolderPath = (collectionFolderPath: string) =>
+    `${collectionFolderPath}/${METADATA_FOLDER_NAME}`;
+
+export const getUniqueFileSaveName = (
+    filename: string,
+    usedFilenamesInCollection: Set<string>
+) => {
+    let fileSaveName = sanitizeName(filename);
+    const count = 1;
+    while (usedFilenamesInCollection.has(fileSaveName)) {
+        const filenameParts = splitFilenameAndExtension(fileSaveName);
+        if (filenameParts[1]) {
+            fileSaveName = `${filenameParts[0]}(${count}).${filenameParts[1]}`;
+        } else {
+            fileSaveName = `${filenameParts[0]}(${count})`;
+        }
+    }
+    usedFilenamesInCollection.add(fileSaveName);
+    return fileSaveName;
+};
+
+export const getFileMetadataSavePath = (
+    collectionFolderPath: string,
+    fileSaveName: string
+) => `${collectionFolderPath}/${METADATA_FOLDER_NAME}/${fileSaveName}.json`;
+
+export const getFileSavePath = (
+    collectionFolderPath: string,
+    fileSaveName: string
+) => `${collectionFolderPath}/${fileSaveName}`;
```

Comment:
Should we handle the case where filename contains invalid characters?
https://stackoverflow.com/a/8485137/546896

Reply:
ya this seems like a more complete check...will update our `sanitizeName` function with this